### ISSUE TITLE
Creating a user with a duplicate name causes a crash

### ISSUE THREAD

**Issue by grantfitzsimmons** [specify/specify7#3920]
<img width="1315" alt="image" src="https://github.com/specify/specify7/assets/37256050/cafa09a1-465b-4adc-919a-b00cb370103b">

```
"Duplicate entry 'abentley' for key 'Name'"
```
```
MySQLdb.IntegrityError: (1062, "Duplicate entry 'abentley' for key 'Name'")
```

**Steps to Recreate:**
1. Create a new user with the same name as an existing user
2. Save the record
3. See error

[Specify 7 Crash Report - 2023-08-16T20 37 47.698Z.txt](https://github.com/specify/specify7/files/12363312/Specify.7.Crash.Report.-.2023-08-16T20.37.47.698Z.txt)

**Reported By:** Carmella at UMass

`v7.8.13` and `v7.9`

**Comment by specifysoftware** [comment]
This issue has been mentioned on **Specify Community Forum**. There might be relevant details there:

https://discourse.specifysoftware.org/t/error-creating-new-agent/1245/6

**Comment by melton-jason** [comment]
This is because there was no uniqueness rule (either on frontend or backend) before the uniqueness synchronizations in v7.9-dev . 
However, this did exist as a constraint in the database. 

Should be fixed in the next release (as of: #2965)

Also see: #2260

**Comment by melton-jason** [comment]
I am closing this for now. The uniqueness rule has been implemented in `v7.9`.

However, when verifying this behavior, I did stumble across a bug. 
The frontend uniqueness rule does its search in a case-sensitive manner while the backend searches and verifies in a case-insensitive way. 

For example, if there is a specifyuser called `spadmin`, the frontend will allow creating a user with name `sPadmin`, but this will thow a backend exception. 
This behavior is most likely not specific to this rule, so I will open a new issue (hence the reasoning behind closing this one)
